src/MarginTrading.SettingsService.Services/MarketDayOffService.cs

```diff
@@ -53,46 +54,120 @@ public class MarketDayOffService : IMarketDayOffService
                     rawPlatformSchedule.WithRank(int.MaxValue)).ToList();
 
                 var tradingDayInfo = GetTradingDayInfo(schedule, currentDateTime);
-                
+
```

Comment:
Also the line above now looks strange. Why are we calculating lastTradingDay & nextTradingDay using the schedule of particular market?